### ISSUE TITLE
_NoopCatFunc in transformer layer

### ISSUE THREAD

**Issue by robot-transformer** [NVIDIA/TransformerEngine#1384]
Hello!
I'm currently rewriting my LM-pipeline with TE. And I noticed that the following [line](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/module/layernorm_linear.py#L1174) doesn't work as expected. I.e for the first forward of my layer this cat noop function goes to [this](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/module/_common.py#L181) point (i.e. all checks of layout were passed), but out tensor instead of being concatenation of all weights looks like cat of tensor[0] and uninitialised memory.
After some debugs I found that tensors[i].data_ptr() + tensors[i].size(dim) * data_ptr_stride = tensors[I+1].data_ptr(), but tensors[i].untyped_storage().data_ptr() are different (but storages located sequentially). And as far as I know here is expected that  tensors[i].untyped_storage().data_ptr() would be the same. Am I wrong?

**Comment by sbhavani** [comment]
This appears to have been fixed in v2.13 by #2664: `_NoopCatFunc` now verifies that the first tensor’s storage covers the full concatenated region and falls back to torch.cat otherwise. 

If you still have problems please create a new issue with latest TE main.
